frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1842,6 +1885,55 @@ class SegmentsView extends React.Component<Props, State> {
     );
   }
 
+  getExpandSubgroupsItem(groupId: number) {
+    const children = this.getSubGroupsAsTreeNodes(groupId);
+    const areAllChildrenExpanded =
+      children.filter((childNode) => this.state.expandedGroupKeys?.includes(childNode)).length ===
+      children.length;
+    const isGroupItselfExpanded = this.state.expandedGroupKeys?.includes(
+      this.getKeyForGroupId(groupId),
+    );
+    if (areAllChildrenExpanded && isGroupItselfExpanded) {
+      return null;
+    }
+    const groupsToExpand = children;
+    // It doesn't make sense to expand subgroups if the group itself is collapsed, so also expand the group.
+    if (!isGroupItselfExpanded) groupsToExpand.push(this.getKeyForGroupId(groupId));
+    return {
+      key: "expandAll",
+      disabled: !this.props.allowUpdate,
```

Comment:
can I expand/collapse a group itself when the annotation is read-only? I think, I would allow this. the save_saga should not save anything, when it's not allowed. so, we don't do any damage (it would be worse to allow mutating actual data and then it is not saved without warning).

Reply:
oh I didnt know that this is what this prop means. makes total sense, I agree that this should be possible.